infrastructure/src/main/java/org/corfudb/infrastructure/FailureHandlerDispatcher.java

```diff
@@ -54,10 +56,10 @@ public void dispatchHandler(IFailureHandlerPolicy failureHandlerPolicy, Layout c
      * Seals the epoch
      * Set local epoch and then attempt to move all servers to new epoch
      *
-     * @param layout Current layout to be sealed
+     * @param currentLayout Current layout to be sealed
      */
-    private void sealEpoch(Layout layout) {
-        layout.setEpoch(layout.getEpoch() + 1);
-        layout.moveServersToEpoch();
+    private void sealEpoch(Layout currentLayout) throws QuorumUnreachableException {
+        currentLayout.setEpoch(currentLayout.getEpoch() + 1);
```

Comment:
Earlier we used to seal the new layout. and we used to increment the epoch of the new layout before sealing. That's why there was no failure earlier.
Now since we need to seal the old one, we do the increment of the old layout here.